### PR TITLE
feat(forge): prompt address and uint cheatcodes

### LINKED ISSUE #5509: Feat(forge): `vm.prompt*` cheatcodes
### Component

Forge

### Describe the feature you would like

Add `vm.promptAddress(string prompt)`, `vm.promptUint(string prompt)`, etc. cheatcodes which would prompt user for some input. This could be used as an alternative of script function args.

Reasons why I think this could be useful:
1. When working on a new external project it would be handy if user could just run the script and follow the instructions after it rather than checking certain function parameters and signatures in random .sol script files to enter them in cli. Also, prompt string would allow to specify more details on what needs to be provided
2. This would allow more straightforward UI because scripts relying on some arguments will not require explicit specification of them in `--sig` argument

Expected behavior:

```solidity
contract Deploy is Script {
    function deployFoo() public {
        address bar = vm.promptAddress("Enter bar address: ");
        new Foo(bar);
    }
}
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamuik16**
Closes #5509 

cc @mattsse 

In case of parse failure, it fails with these errors:

```bash
[FAIL. Reason: failed parsing "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293B" as type `address`: parser error:
0x3C44CdDdB6a900fa2b585dd299e03d12FA4293B
^
Odd number of digits] testPrompt_Address() (gas: 2948)
```
```bash 
[FAIL. Reason: failed parsing "-1" as type `uint256`: parser error:
-1
^
expected at least one digit] testPrompt_Uint() (gas: 3008)
```

### CODE REVIEW

File: crates/cheatcodes/src/fs.rs [hunk 1]
```diff
@@ -374,6 +376,20 @@ impl Cheatcode for promptSecretCall {
     }
 }
 
+impl Cheatcode for promptAddressCall {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { promptText: text } = self;
+        parse(prompt(state, text, prompt_input).unwrap().as_str(), &DynSolType::Address)
```
Comment (DaniPopes) [inline crates/cheatcodes/src/fs.rs:382]:
```suggestion
        parse(&prompt(state, text, prompt_input)?, &DynSolType::Address)
```

File: crates/cheatcodes/src/fs.rs [hunk 2]
```diff
@@ -374,6 +376,20 @@ impl Cheatcode for promptSecretCall {
...
+    }
+}
+
+impl Cheatcode for promptUintCall {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { promptText: text } = self;
+        parse(prompt(state, text, prompt_input).unwrap().as_str(), &DynSolType::Uint(256))
```
Comment (DaniPopes) [inline crates/cheatcodes/src/fs.rs:389]:
```suggestion
        parse(&prompt(state, text, prompt_input)?, &DynSolType::Uint(256))
```

File: testdata/default/cheats/Prompt.t.sol [hunk 3]
```diff
@@ -15,4 +16,14 @@ contract PromptTest is DSTest {
...
+
+    function testPrompt_Address() public {
+        address test = vm.promptAddress("test");
+        console.log(test);
+    }
+
+    function testPrompt_Uint() public {
+        uint256 test = vm.promptUint("test");
+        console.log(test);
+    }
```
Comment (Evalir) [inline testdata/default/cheats/Prompt.t.sol:28]: let's improve these tests, and assert that what gets parsed is actually what we expect

**Review by DaniPopes — COMMENTED** [review]
I'm not sure about this, since these are just `vm.parse<Type>(vm.prompt())` users can do this in "userland", rather than keep bloating the already massive list of cheatcodes we have.

cc @mattsse @mds1 @klkvr

**Review by Evalir — CHANGES_REQUESTED** [review]
some feedback on tests:

**Review by mattsse — APPROVED** [review]
I think the most common prompts will be address and uint, so having a builtin for these makes sense imo

File: testdata/default/cheats/Prompt.t.sol (hunk 3 above)
Comment (kamuik16) [reply testdata/default/cheats/Prompt.t.sol]: added assertEq with what I am prompting as an input.

**Comment by mds1** [comment]
One argument in favor of native cheats here is better UX. If I do `vm.parseAddress(vm.prompt())` and the user leaves a character off the address, the script would revert when parsing fails, and now you have to start over. But if I do `vm.promptAddress()` I'd expect the TUI to say "invalid address" and let you try again without starting over

**Comment by kamuik16** [comment]
> One argument in favor of native cheats here is better UX. If I do `vm.parseAddress(vm.prompt())` and the user leaves a character off the address, the script would revert when parsing fails, and now you have to start over. But if I do `vm.promptAddress()` I'd expect the TUI to say "invalid address" and let you try again without starting over

We can definitely add this. We can either ask the user for input 2 times and if both the times fail, it will exit or we can run while loop till we get a proper input from the user. What would be the ideal way to do this?

**Review by mattsse — APPROVED** [review]

**Pull request merged**